src/runtime-rs/crates/hypervisor/src/qemu/qmp.rs

```diff
@@ -491,6 +491,43 @@ impl Qmp {
 
         Err(anyhow!("no target device found"))
     }
+
+    pub fn hotplug_vfio_device(
+        &mut self,
+        primary_device: &mut HostDevice,
```

Comment:
Just like in #11144 (and others IIRC), I'd really prefer not to make the qmp module depend on `HostDevice`.  Seeing as just `bus_slot_func` and `hostdev_id` are actually used, could we just pass in those instead?  We can then return `guest_pci_path` and assign it to `HostDevice` in the caller in `inner.rs` which depends on `HostDevice` already anyway.

Reply:
sure, I will update it.